homeassistant/components/solarlog/sensor.py

```diff
@@ -0,0 +1,233 @@
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = "solar-log"
+DEFAULT_NAME = "solarlog"
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=60)
+
+"""Supported sensor types."""
+SENSOR_TYPES = {
+    "time": ["TIME", "last update", None, "mdi:solar-power"],
+    "power_ac": ["powerAC", "power AC", POWER_WATT, "mdi:solar-power"],
+    "power_dc": ["powerDC", "power DC", POWER_WATT, "mdi:solar-power"],
+    "voltage_ac": ["voltageAC", "voltage AC", "V", "mdi:solar-power"],
+    "voltage_dc": ["voltageDC", "voltage DC", "V", "mdi:solar-power"],
+    "yield_day": ["yieldDAY", "yield day", ENERGY_KILO_WATT_HOUR, "mdi:solar-power"],
+    "yield_yesterday": [
+        "yieldYESTERDAY",
+        "yield yesterday",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "yield_month": [
+        "yieldMONTH",
+        "yield month",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "yield_year": ["yieldYEAR", "yield year", ENERGY_KILO_WATT_HOUR, "mdi:solar-power"],
+    "yield_total": [
+        "yieldTOTAL",
+        "yield total",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "consumption_ac": [
+        "consumptionAC",
+        "consumption AC",
+        POWER_WATT,
+        "mdi:solar-power",
+    ],
+    "consumption_day": [
+        "consumptionDAY",
+        "consumption day",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "consumption_yesterday": [
+        "consumptionYESTERDAY",
+        "consumption yesterday",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "consumption_month": [
+        "consumptionMONTH",
+        "consumption month",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "consumption_year": [
+        "consumptionYEAR",
+        "consumption year",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "consumption_total": [
+        "consumptionTOTAL",
+        "consumption total",
+        ENERGY_KILO_WATT_HOUR,
+        "mdi:solar-power",
+    ],
+    "total_power": ["totalPOWER", "total power", "Wp", "mdi:solar-power"],
+    "alternator_loss": [
+        "alternatorLOSS",
+        "alternator loss",
+        POWER_WATT,
+        "mdi:solar-power",
+    ],
+    "capacity": ["CAPACITY", "capacity", "%", "mdi:solar-power"],
+    "efficiency": ["EFFICIENCY", "efficiency", "% W/Wp", "mdi:solar-power"],
+    "power_available": [
+        "powerAVAILABLE",
+        "power available",
+        POWER_WATT,
+        "mdi:solar-power",
+    ],
+    "usage": ["USAGE", "usage", None, "mdi:solar-power"],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): cv.time_period,
```

Comment:
Let's not make the scan interval configurable, instead let's just set it to a reasonable default.

Reply:
fixed. I've removed the config option and made a fixed scan interval of 1 minute.